packetbeat/beater/processor.go

```diff
@@ -111,17 +125,37 @@ func newProcessorFactory(name string, err chan error, beat *beat.Beat, configura
 	}
 }
 
+// / CreateWithReporter functions the same as Create, but also accepts a StatusReporter.
+func (p *processorFactory) CreateWithReporter(pipeline beat.PipelineConnector, cfg *conf.C, statusReporter status.StatusReporter) (cfgfile.Runner, error) {
+	if statusReporter != nil {
+		statusReporter.UpdateStatus(status.Configuring, "starting packetbeat processor configuration")
+	}
+	duration, publisher, flows, sniffer, errChan, err := p.create(pipeline, cfg, statusReporter)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create packetbeat processor: %w", err)
+	}
+	return newProcessor(duration, publisher, flows, sniffer, errChan, statusReporter), nil
+}
+
 // Create returns a new module runner that publishes to the provided pipeline, configured from cfg.
 func (p *processorFactory) Create(pipeline beat.PipelineConnector, cfg *conf.C) (cfgfile.Runner, error) {
```

Comment:
Looks like this could now be a one-liner that just calls `CreateWithReporter` with a nil `statusReporter`